src/Config/Factory/Cache.php

```diff
@@ -115,8 +124,14 @@ public function get(): array
 
         return $this->configMerger->mergeConfigs([
             'frontend' => [
-                'default' => $redisCache,
-                'page_cache' => $redisCache,
+                'default' => array_replace_recursive(
```

Comment:
You can change `$redisCache` to closure with database id as an argument.
It will be more readable 
```
'default' => $redisCache(self::REDIS_DATABASE_DEFAULT),
'page_cache' => $redisCache(self::REDIS_DATABASE_PAGE_CACHE),
```

Reply:
I tried, but then I need to add an additional variable

Reply:
Why this a problem?
Just change
```
$redisCache = [
            'backend' => 'Cm_Cache_Backend_Redis',
            'backend_options' => [
                'server' => $redisConfig[0]['host'],
                'port' => $redisConfig[0]['port'],
                'database' => 1,
            ],
        ];
```
to
```
$redisCache = function($databaseId = 1) use ($redisConfig) {
      return [
            'backend' => 'Cm_Cache_Backend_Redis',
            'backend_options' => [
                'server' => $redisConfig[0]['host'],
                'port' => $redisConfig[0]['port'],
                'database' => $databaseId,
            ],
        ];
}
```
And then use
```
$redisCache(self::REDIS_DATABASE_DEFAULT)
```
instead of
```
array_replace_recursive(
       $redisCache,
       ['backend_options' => ['database' => self::REDIS_DATABASE_DEFAULT]]
 ),
```
For me it looks more clear.

Reply:
I think it will be better if we leave as is.
In line 112, we change Redis configuration if we use a slave connection.
https://github.com/magento/ece-tools/pull/440/files#diff-e630518a0378691f81d51f6b3fc17cc8R112